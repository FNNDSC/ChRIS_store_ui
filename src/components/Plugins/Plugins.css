.plugins-stats {
  /* color */
  background: #f5f5f5;
  background: var(--pf-black-150);
  
  /* border */
  border-bottom-color: #bbbbbb;
  border-bottom: 1px solid var(--pf-black-400);
}

.plugins-stats-row {
  /* display */
  display: flex;

  /* position */
  position: relative;
}
 
.plugins-found {
  /* color */
  color: #393f44;
  color: var(--pf-black-800);
  
  /* display */
  padding: 1em;

  /* font */
  font-size: 1.5em;
  font-weight: 600;
}

.dropdown.btn-group {
  /* display */
  height: 26px; 
  margin: 1.65em 1em 1.65em;

  /* position */
  position: absolute;
  right: 0;
}

.plugins-list {
  /* display */
  padding: 2em;
  
  /* flex */
  flex: 3;
}

@media (min-width: 768px)  {
  .plugins-list {
    /* display */
    width: 0;
    padding: 0 3em 2em;
  }
}
