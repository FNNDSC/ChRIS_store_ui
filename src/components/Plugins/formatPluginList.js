const formatItemData = itemData => itemData.reduce((sum, pair) => {
  const result = {};
  result[pair.name] = pair.value;
  return Object.assign(sum, result);
}, {});

const formatPluginList = pluginList => pluginList.map((item) => {
  const result = Object.assign({}, item);
  result.data = formatItemData(result.data);
  return result;
});

export default formatPluginList;
