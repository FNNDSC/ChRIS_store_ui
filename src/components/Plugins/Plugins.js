import React, { Component } from 'react';
import { StoreClient } from '@fnndsc/chrisstoreapi';
import { DropdownButton, MenuItem } from 'react-bootstrap';
import PluginItem from './components/PluginItem/PluginItem';
import LoadingPluginItem from './components/LoadingPluginItem/LoadingPluginItem';
import PluginsCategories from './components/PluginsCategories/PluginsCategories';
import './Plugins.css';
import LoadingContainer from '../LoadingContainer/LoadingContainer';
import LoadingContent from '../LoadingContainer/components/LoadingContent/LoadingContent';

// ==============================
// ------ PLUGINS COMPONENT -----
// ==============================

class Plugins extends Component {
  constructor() {
    super();
    this.state = {
      pluginList: null,
      categories: [
        {
          name: 'Visualization',
          length: 3,
        },
        {
          name: 'Modeling',
          length: 11,
        },
        {
          name: 'Statistical Operation',
          length: 7,
        },
      ],
    };

    this.fetchPlugins = this.fetchPlugins.bind(this);
  }

  componentDidMount() {
    this.fetchPlugins().catch((err) => {
      console.error(err);
    });
  }

  fetchPlugins() {
    const storeURL = process.env.REACT_APP_STORE_URL;
    const authURL = process.env.REACT_APP_STORE_AUTH_URL;

    return new Promise(async (resolve, reject) => {
      let plugins;
      const searchParams = {
        limit: 20,
        offset: 0,
      };

      try {
        const token = await StoreClient.getAuthToken(authURL, 'cube', 'cube1234');
        const client = new StoreClient(storeURL, { token });

        // add plugins to pluginList as they are received
        plugins = await client.getPlugins(searchParams, (onePageResponse) => {
          const onePagePlugins = onePageResponse.plugins;

          this.setState((prevState) => {
            const prevPluginList = prevState.pluginList ? prevState.pluginList : [];
            const nextPluginList = prevPluginList.concat(onePagePlugins);
            return { pluginList: nextPluginList };
          });
        });
      } catch (e) {
        return reject(e);
      }
      return resolve(plugins);
    });
  }

  render() {
    const { pluginList, categories } = this.state;
    let pluginListBody;

    // Remove email from author
    const removeEmail = author => author.replace(/( ?\(.*\))/g, '');

    // Render the pluginList if the plugins have been fetched
    if (pluginList) {
      pluginListBody = pluginList.map(plugin => (
        <PluginItem
          title={plugin.title}
          name={plugin.name}
          author={removeEmail(plugin.authors)}
          creationDate={plugin.creation_date}
          key={plugin.dock_image}
        />
      ));
    // Or else show the loading text
    } else {
      pluginListBody = new Array(6).fill().map((e, i) => (
        // eslint-disable-next-line react/no-array-index-key
        <LoadingPluginItem key={i} />
      ));
    }

    let pluginsFound;
    if (pluginList) {
      pluginsFound = (
        <span className="plugins-found">{pluginList.length} plugins found</span>
      );
    } else {
      pluginsFound = (
        <LoadingContainer>
          <LoadingContent
            width="135px"
            height="30px"
            left="1em"
            top="1.5em"
            bottom="1.5em"
          />
        </LoadingContainer>
      );
    }

    return (
      <div className="plugins-container">
        <div className="plugins-stats">
          <div className="row plugins-stats-row">
            { pluginsFound }
            <DropdownButton
              id="sort-by-dropdown"
              title="Sort By"
              pullRight
            >
              <MenuItem eventKey="1">Name</MenuItem>
            </DropdownButton>
          </div>
        </div>
        <div className="row plugins-row">
          <PluginsCategories categories={categories} />
          <div className="plugins-list">
            {pluginListBody}
          </div>
        </div>
      </div>
    );
  }
}

export default Plugins;
