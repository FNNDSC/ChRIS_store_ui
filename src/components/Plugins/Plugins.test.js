import React from 'react';
import { shallow } from 'enzyme';
import Plugins from './Plugins';

// define mock for @fnndsc/chrisstoreapi module
jest.mock('@fnndsc/chrisstoreapi', () => require.requireActual('../__mocks__/chrisstoreapi').default);

describe('Plugins', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Plugins />);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render plugins-container div', () => {
    expect(wrapper.find('div.plugins-container')).toHaveLength(1);
  });

  it('should render plugins-stats inside of plugins-container', () => {
    expect(wrapper
      .find('div.plugins-container')
      .find('div.plugins-stats'))
      .toHaveLength(1);
  });

  it('should render row div inside plugins-stats', () => {
    expect(wrapper
      .find('div.plugins-stats')
      .find('div.row'))
      .toHaveLength(1);
  });

  it('should render LoadingContainer component inside plugins-stats row', () => {
    expect(wrapper
      .find('div.plugins-stats-row')
      .find('LoadingContainer'))
      .toHaveLength(1);
  });

  it('should render LoadingContent component inside LoadingContainer', () => {
    expect(wrapper
      .find('div.plugins-stats-row')
      .find('LoadingContainer')
      .find('LoadingContent'))
      .toHaveLength(1);
  });

  it('should render DropdownButton component inside plugins-stats row', () => {
    expect(wrapper
      .find('div.plugins-stats-row')
      .find('DropdownButton'))
      .toHaveLength(1);
  });

  it('should render 1 MenuItem component inside DropdownButton', () => {
    expect(wrapper
      .find('DropdownButton')
      .find('MenuItem'))
      .toHaveLength(1);
  });

  it('every MenuItem component should have eventKey prop', () => {
    wrapper.find('MenuItem').forEach((item) => {
      expect(item.prop('eventKey')).toBeDefined();
    });
  });

  it('should render plugins-row inside of plugins-container', () => {
    expect(wrapper
      .find('div.plugins-container')
      .find('div.plugins-row.row').length)
      .toEqual(1);
  });

  it('should render PluginsCategories component inside plugins-row', () => {
    expect(wrapper
      .find('div.plugins-row')
      .find('PluginsCategories'))
      .toHaveLength(1);
  });

  it('should render plugins-list div inside plugins-row', () => {
    expect(wrapper
      .find('div.plugins-row')
      .find('div.plugins-list').length)
      .toEqual(1);
  });

  it('should render 6 LoadingPluginItem components if pluginList is null', () => {
    expect(wrapper
      .find('div.plugins-list')
      .find('LoadingPluginItem'))
      .toHaveLength(6);
  });

  /* ============================== */
  /* =========== METHODS ========== */
  /* ============================== */

  it('sets mounted to true on after mounting', () => {
    expect(Plugins.prototype.mounted).toBeFalsy();
    Plugins.prototype.componentWillMount();
    expect(Plugins.prototype.mounted).toBeTruthy();
  });

  it('sets mounted to false after componentWillUnmount is called', () => {
    const instance = wrapper.instance();
    expect(instance.mounted).toBeTruthy();
    instance.componentWillUnmount();
    expect(instance.mounted).toBeFalsy();
  });

  /* ============================== */
  /* ====== FETCH PLUGINS FN ====== */
  /* ============================== */

  it('should have fetchPlugins function', () => {
    const { fetchPlugins } = wrapper.instance();
    expect(fetchPlugins).toBeDefined();
  });

  it('should be able to fetch plugins', () => {
    const { fetchPlugins } = wrapper.instance();
    return fetchPlugins().then((plugins) => {
      expect(plugins).toBeDefined();
    });
  });
});

const samplePluginList = [
  {
    title: 'testTitle1',
    id: 1,
    name: 'testName1',
    authors: 'testAuthor1',
    dock_image: 'dock/image1',
    creation_date: '2018-06-19T15:29:11.349272Z',
  },
  {
    title: 'testTitle2',
    id: 2,
    name: 'testName2',
    authors: 'testAuthor2',
    dock_image: 'dock/image2',
    creation_date: '2018-06-19T15:29:11.349272Z',
  },
];

describe('rendered Plugins', () => {
  let wrapper;
  let plugins;
  beforeEach(() => {
    wrapper = shallow(<Plugins />);
    wrapper.setState({ pluginList: samplePluginList });
    plugins = Array.from(wrapper.find('Plugin'));
  });

  it('should render the correct number of Plugin components', () => {
    expect(wrapper.find('Plugin')).toHaveLength(2);
  });

  it('should render the correct number for "plugins found"', () => {
    expect(wrapper.find('span.plugins-found').text()).toEqual('2 plugins found');
  });

  it('should render the title prop for each Plugin', () => {
    plugins.forEach((plugin, i) => {
      expect(plugin.props.title).toEqual(`testTitle${i + 1}`);
    });
  });

  it('should render the name prop for each Plugin', () => {
    plugins.forEach((plugin, i) => {
      expect(plugin.props.name).toEqual(`testName${i + 1}`);
    });
  });

  it('should render the author prop for each Plugin', () => {
    plugins.forEach((plugin, i) => {
      expect(plugin.props.author).toEqual(`testAuthor${i + 1}`);
    });
  });

  it('should render the creation_date prop for each Plugin', () => {
    plugins.forEach((plugin) => {
      expect(plugin.props.creationDate).toEqual('2018-06-19T15:29:11.349272Z');
    });
  });
});
