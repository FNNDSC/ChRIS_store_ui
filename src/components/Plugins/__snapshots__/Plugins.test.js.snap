// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Plugins should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Plugins />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="plugins-stats"
>
          <div
                    className="row plugins-stats-row"
          >
                    <LoadingContainer
                              className=""
                    >
                              <LoadingContent
                                        bottom="1.5em"
                                        className=""
                                        height="30px"
                                        left="1em"
                                        right="0"
                                        top="1.5em"
                                        type=""
                                        width="135px"
                              />
                    </LoadingContainer>
                    <DropdownButton
                              id="sort-by-dropdown"
                              pullRight={true}
                              title="Sort By"
                    >
                              <MenuItem
                                        bsClass="dropdown"
                                        disabled={false}
                                        divider={false}
                                        eventKey="1"
                                        header={false}
                              >
                                        Name
                              </MenuItem>
                    </DropdownButton>
          </div>
</div>,
        <div
          className="row plugins-row"
>
          <PluginsCategories
                    categories={
                              Array [
                                        Object {
                                          "length": 3,
                                          "name": "Visualization",
                                        },
                                        Object {
                                          "length": 11,
                                          "name": "Modeling",
                                        },
                                        Object {
                                          "length": 7,
                                          "name": "Statistical Operation",
                                        },
                                      ]
                    }
          />
          <div
                    className="plugins-list"
          >
                    <LoadingPluginItem />
                    <LoadingPluginItem />
                    <LoadingPluginItem />
                    <LoadingPluginItem />
                    <LoadingPluginItem />
                    <LoadingPluginItem />
          </div>
</div>,
      ],
      "className": "plugins-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="row plugins-stats-row"
>
            <LoadingContainer
                        className=""
            >
                        <LoadingContent
                                    bottom="1.5em"
                                    className=""
                                    height="30px"
                                    left="1em"
                                    right="0"
                                    top="1.5em"
                                    type=""
                                    width="135px"
                        />
            </LoadingContainer>
            <DropdownButton
                        id="sort-by-dropdown"
                        pullRight={true}
                        title="Sort By"
            >
                        <MenuItem
                                    bsClass="dropdown"
                                    disabled={false}
                                    divider={false}
                                    eventKey="1"
                                    header={false}
                        >
                                    Name
                        </MenuItem>
            </DropdownButton>
</div>,
          "className": "plugins-stats",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <LoadingContainer
                className=""
>
                <LoadingContent
                                bottom="1.5em"
                                className=""
                                height="30px"
                                left="1em"
                                right="0"
                                top="1.5em"
                                type=""
                                width="135px"
                />
</LoadingContainer>,
              <DropdownButton
                id="sort-by-dropdown"
                pullRight={true}
                title="Sort By"
>
                <MenuItem
                                bsClass="dropdown"
                                disabled={false}
                                divider={false}
                                eventKey="1"
                                header={false}
                >
                                Name
                </MenuItem>
</DropdownButton>,
            ],
            "className": "row plugins-stats-row",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <LoadingContent
                  bottom="1.5em"
                  className=""
                  height="30px"
                  left="1em"
                  right="0"
                  top="1.5em"
                  type=""
                  width="135px"
/>,
                "className": "",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "bottom": "1.5em",
                  "className": "",
                  "height": "30px",
                  "left": "1em",
                  "right": "0",
                  "top": "1.5em",
                  "type": "",
                  "width": "135px",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <MenuItem
                  bsClass="dropdown"
                  disabled={false}
                  divider={false}
                  eventKey="1"
                  header={false}
>
                  Name
</MenuItem>,
                "id": "sort-by-dropdown",
                "pullRight": true,
                "title": "Sort By",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "dropdown",
                  "children": "Name",
                  "disabled": false,
                  "divider": false,
                  "eventKey": "1",
                  "header": false,
                },
                "ref": null,
                "rendered": "Name",
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <PluginsCategories
              categories={
                            Array [
                                          Object {
                                            "length": 3,
                                            "name": "Visualization",
                                          },
                                          Object {
                                            "length": 11,
                                            "name": "Modeling",
                                          },
                                          Object {
                                            "length": 7,
                                            "name": "Statistical Operation",
                                          },
                                        ]
              }
/>,
            <div
              className="plugins-list"
>
              <LoadingPluginItem />
              <LoadingPluginItem />
              <LoadingPluginItem />
              <LoadingPluginItem />
              <LoadingPluginItem />
              <LoadingPluginItem />
</div>,
          ],
          "className": "row plugins-row",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "categories": Array [
                Object {
                  "length": 3,
                  "name": "Visualization",
                },
                Object {
                  "length": 11,
                  "name": "Modeling",
                },
                Object {
                  "length": 7,
                  "name": "Statistical Operation",
                },
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <LoadingPluginItem />,
                <LoadingPluginItem />,
                <LoadingPluginItem />,
                <LoadingPluginItem />,
                <LoadingPluginItem />,
                <LoadingPluginItem />,
              ],
              "className": "plugins-list",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "0",
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "1",
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "2",
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "3",
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "4",
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "5",
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="plugins-stats"
>
            <div
                        className="row plugins-stats-row"
            >
                        <LoadingContainer
                                    className=""
                        >
                                    <LoadingContent
                                                bottom="1.5em"
                                                className=""
                                                height="30px"
                                                left="1em"
                                                right="0"
                                                top="1.5em"
                                                type=""
                                                width="135px"
                                    />
                        </LoadingContainer>
                        <DropdownButton
                                    id="sort-by-dropdown"
                                    pullRight={true}
                                    title="Sort By"
                        >
                                    <MenuItem
                                                bsClass="dropdown"
                                                disabled={false}
                                                divider={false}
                                                eventKey="1"
                                                header={false}
                                    >
                                                Name
                                    </MenuItem>
                        </DropdownButton>
            </div>
</div>,
          <div
            className="row plugins-row"
>
            <PluginsCategories
                        categories={
                                    Array [
                                                Object {
                                                  "length": 3,
                                                  "name": "Visualization",
                                                },
                                                Object {
                                                  "length": 11,
                                                  "name": "Modeling",
                                                },
                                                Object {
                                                  "length": 7,
                                                  "name": "Statistical Operation",
                                                },
                                              ]
                        }
            />
            <div
                        className="plugins-list"
            >
                        <LoadingPluginItem />
                        <LoadingPluginItem />
                        <LoadingPluginItem />
                        <LoadingPluginItem />
                        <LoadingPluginItem />
                        <LoadingPluginItem />
            </div>
</div>,
        ],
        "className": "plugins-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="row plugins-stats-row"
>
              <LoadingContainer
                            className=""
              >
                            <LoadingContent
                                          bottom="1.5em"
                                          className=""
                                          height="30px"
                                          left="1em"
                                          right="0"
                                          top="1.5em"
                                          type=""
                                          width="135px"
                            />
              </LoadingContainer>
              <DropdownButton
                            id="sort-by-dropdown"
                            pullRight={true}
                            title="Sort By"
              >
                            <MenuItem
                                          bsClass="dropdown"
                                          disabled={false}
                                          divider={false}
                                          eventKey="1"
                                          header={false}
                            >
                                          Name
                            </MenuItem>
              </DropdownButton>
</div>,
            "className": "plugins-stats",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <LoadingContainer
                  className=""
>
                  <LoadingContent
                                    bottom="1.5em"
                                    className=""
                                    height="30px"
                                    left="1em"
                                    right="0"
                                    top="1.5em"
                                    type=""
                                    width="135px"
                  />
</LoadingContainer>,
                <DropdownButton
                  id="sort-by-dropdown"
                  pullRight={true}
                  title="Sort By"
>
                  <MenuItem
                                    bsClass="dropdown"
                                    disabled={false}
                                    divider={false}
                                    eventKey="1"
                                    header={false}
                  >
                                    Name
                  </MenuItem>
</DropdownButton>,
              ],
              "className": "row plugins-stats-row",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": <LoadingContent
                    bottom="1.5em"
                    className=""
                    height="30px"
                    left="1em"
                    right="0"
                    top="1.5em"
                    type=""
                    width="135px"
/>,
                  "className": "",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "bottom": "1.5em",
                    "className": "",
                    "height": "30px",
                    "left": "1em",
                    "right": "0",
                    "top": "1.5em",
                    "type": "",
                    "width": "135px",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <MenuItem
                    bsClass="dropdown"
                    disabled={false}
                    divider={false}
                    eventKey="1"
                    header={false}
>
                    Name
</MenuItem>,
                  "id": "sort-by-dropdown",
                  "pullRight": true,
                  "title": "Sort By",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "bsClass": "dropdown",
                    "children": "Name",
                    "disabled": false,
                    "divider": false,
                    "eventKey": "1",
                    "header": false,
                  },
                  "ref": null,
                  "rendered": "Name",
                  "type": [Function],
                },
                "type": [Function],
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <PluginsCategories
                categories={
                                Array [
                                                Object {
                                                  "length": 3,
                                                  "name": "Visualization",
                                                },
                                                Object {
                                                  "length": 11,
                                                  "name": "Modeling",
                                                },
                                                Object {
                                                  "length": 7,
                                                  "name": "Statistical Operation",
                                                },
                                              ]
                }
/>,
              <div
                className="plugins-list"
>
                <LoadingPluginItem />
                <LoadingPluginItem />
                <LoadingPluginItem />
                <LoadingPluginItem />
                <LoadingPluginItem />
                <LoadingPluginItem />
</div>,
            ],
            "className": "row plugins-row",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "categories": Array [
                  Object {
                    "length": 3,
                    "name": "Visualization",
                  },
                  Object {
                    "length": 11,
                    "name": "Modeling",
                  },
                  Object {
                    "length": 7,
                    "name": "Statistical Operation",
                  },
                ],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <LoadingPluginItem />,
                  <LoadingPluginItem />,
                  <LoadingPluginItem />,
                  <LoadingPluginItem />,
                  <LoadingPluginItem />,
                  <LoadingPluginItem />,
                ],
                "className": "plugins-list",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "0",
                  "nodeType": "function",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": "1",
                  "nodeType": "function",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": "2",
                  "nodeType": "function",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": "3",
                  "nodeType": "function",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": "4",
                  "nodeType": "function",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": "5",
                  "nodeType": "function",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
